Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -354,13 +524,53 @@ bool isFunctionJson(string fullName)
             }
         }
 
-        private async Task<string> DeployZipLocally(ZipDeploymentInfo zipDeploymentInfo, ITracer tracer)
+        private async Task ArtifactFetch(IRepository repository, DeploymentInfoBase deploymentInfo, string targetBranch, ILogger logger, ITracer tracer)
+        {
+            var artifactDeploymentInfo = (ArtifactDeploymentInfo)deploymentInfo;
+            //Fetches content from URL if ArtifactURL exists
+            HttpContent content = !string.IsNullOrEmpty(artifactDeploymentInfo.ArtifactURL)
+                ? await DeploymentHelper.GetArtifactContentFromURL(artifactDeploymentInfo, tracer)
+                : Request.Content;
+
+            //repository.RepositoryPath is the directory the deployment pipeline expects the files that will be deployed to be
+            var tempDirPath = repository.RepositoryPath;
+            if (deploymentInfo.CleanupTargetDirectory)
```

Comment:
cleanupTargetDirectory is for cleaning up the final target. But looks like you are using it to decide whether or not to clean up the temp directory?